crypto/dso/dso_dlfcn.c

```diff
@@ -308,6 +308,77 @@ static int dladdr(void *address, Dl_info *dl)
 }
 # endif                         /* __sgi */
 
+# ifdef _AIX
+/*-
+ * See IBM's AIX Version 7.2, Technical Reference:
+ *  Base Operating System and Extensions, Volume 1 and 2
+ *  https://www.ibm.com/support/knowledgecenter/ssw_aix_72/com.ibm.aix.base/technicalreferences.htm
+ */
+#  include <sys/ldr.h>
+#  include <errno.h>
+/* ~ 64 * (sizeof(struct ld_info) + _XOPEN_PATH_MAX + _XOPEN_NAME_MAX) */
+#  define DLFCN_LDINFO_SIZE 86976
+typedef struct Dl_info {
+    const char *dli_fname;
+} Dl_info;
+/*
+ * This dladdr()-implementation will also find the ptrgl (Pointer Glue) virtual
+ * address of a function, which is just located in the DATA segment instead of
+ * the TEXT segment.
+ */
+static int dladdr(void *addr, Dl_info *dl)
+{
+    unsigned int found = 0;
+    struct ld_info *ldinfos, *next_ldi, *this_ldi;
+
+    if ((ldinfos = (struct ld_info *)OPENSSL_malloc(DLFCN_LDINFO_SIZE)) == NULL) {
+        errno = ENOMEM;
+        dl->dli_fname = NULL;
+        return 0;
+    }
+
+    if ((loadquery(L_GETINFO, (void *)ldinfos, DLFCN_LDINFO_SIZE)) < 0) {
+        /*-
+         * Error handling is done through errno and dlerror() reading errno:
+         *  ENOMEM (ldinfos buffer is too small),
+         *  EINVAL (invalid flags),
+         *  EFAULT (invalid ldinfos ptr)
+         */
+        OPENSSL_free((void *)ldinfos);
+        dl->dli_fname = NULL;
+        return 0;
+    }
+    next_ldi = ldinfos;
+
+    do {
+        this_ldi = next_ldi;
+        if ((((uintptr_t)addr >= (uintptr_t)this_ldi->ldinfo_textorg)
+             && ((uintptr_t)addr <
+                 ((uintptr_t)this_ldi->ldinfo_textorg +
+                     this_ldi->ldinfo_textsize)))
+            || (((uintptr_t)addr >= (uintptr_t)this_ldi->ldinfo_dataorg)
+                && ((uintptr_t)addr <
+                    ((uintptr_t)this_ldi->ldinfo_dataorg +
+                        this_ldi->ldinfo_datasize)))) {
```

Comment:
I wonder what would people think of following formatting instead
```
        if ((((uintptr_t)addr >= (uintptr_t)this_ldi->ldinfo_textorg)
             && ((uintptr_t)addr < ((uintptr_t)this_ldi->ldinfo_textorg +
                                    this_ldi->ldinfo_textsize)))
            || (((uintptr_t)addr >= (uintptr_t)this_ldi->ldinfo_dataorg)
                && ((uintptr_t)addr < ((uintptr_t)this_ldi->ldinfo_dataorg +
                                       this_ldi->ldinfo_datasize)))) {
```

Reply:
Yes, that does look better

Reply:
Another way to improve readability would be to to declare addr as uintptr_t. I mean 
```
static int dladdr(void *ptr, Dl_info *dl)
{
    uintptr_t addr = (uintptr_t)ptr;
    ...
```

Reply:
Yes, that does look far better. Commit follows ...